src/OrleansAWSUtils/Streams/SQSAdapter.cs

```diff
@@ -0,0 +1,56 @@
+﻿using Orleans.Streams;
+using OrleansAWSUtils.Storage;
+using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+
+namespace OrleansAWSUtils.Streams
+{
+    internal class SQSAdapter : IQueueAdapter
+    {
+        protected readonly string DeploymentId;
+        protected readonly string DataConnectionString;
+        private readonly HashRingBasedStreamQueueMapper streamQueueMapper;
+        protected readonly ConcurrentDictionary<QueueId, SQSStorage> Queues = new ConcurrentDictionary<QueueId, SQSStorage>();
+
+        public string Name { get; private set; }
+        public bool IsRewindable { get { return false; } }
+
+        public StreamProviderDirection Direction { get { return StreamProviderDirection.ReadWrite; } }
+
+        public SQSAdapter(HashRingBasedStreamQueueMapper streamQueueMapper, string dataConnectionString, string deploymentId, string providerName)
+        {
+            if (string.IsNullOrEmpty(dataConnectionString)) throw new ArgumentNullException("dataConnectionString");
+            if (string.IsNullOrEmpty(deploymentId)) throw new ArgumentNullException("deploymentId");
+
+            DataConnectionString = dataConnectionString;
+            DeploymentId = deploymentId;
+            Name = providerName;
+            this.streamQueueMapper = streamQueueMapper;
+        }
+
+        public IQueueAdapterReceiver CreateReceiver(QueueId queueId)
+        {
+            return SQSAdapterReceiver.Create(queueId, DataConnectionString, DeploymentId);
+        }
+
+        public async Task QueueMessageBatchAsync<T>(Guid streamGuid, string streamNamespace, IEnumerable<T> events, StreamSequenceToken token, Dictionary<string, object> requestContext)
+        {
+            if (token != null)
+            {
+                throw new ArgumentException("AzureQueue stream provider currebtly does not support non-null StreamSequenceToken.", "token");
```

Comment:
Ops... The implementation is 99% the same as Azure, so copy and paste miss that...